lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -774,6 +811,52 @@ defmodule Kernel.ParallelCompiler do
     {{:error, Enum.reverse(errors, fun.()), info}, state}
   end
 
+  defp load_pending(kind, module, result, state) do
+    case result do
+      %{{:module, ^module} => {binary, load_status}}
+      when kind in [:module, :struct] and is_binary(binary) ->
+        case load_status do
+          true ->
+            {:found, result}
+
+          false ->
+            pid = load_module(module, binary, state.dest)
+            result = Map.put(result, {:module, module}, {binary, pid})
+            {{:loading, pid}, result}
+
+          pid when is_pid(pid) ->
+            {{:loading, pid}, result}
+        end
+
+      _ ->
+        {:found, result}
+    end
+  end
+
+  defp load_module(module, binary, dest) do
```

Comment:
```suggestion
  # We load modules in a separate process to avoid blocking
  # the parallel compiler. We store the PID of this process and
  # all entries monitor it to know once the module is loaded.
  defp load_module(module, binary, dest) do
```